[docs] Add docs for new EAS functions (#27062)
expo/expo: docs/pages/custom-builds/schema.mdx

  href="https://github.com/expo/eas-build/blob/main/packages/build-tools/src/steps/functions/uploadArtifact.ts"
/>

#### `eas/install_maestro`

> Makes sure [Maestro](https://maestro.mobile.dev), the mobile UI testing framework, is installed, along with all its dependencies. 

Reviewer: ```suggestion
Make sure [Maestro](https://maestro.mobile.dev), the mobile UI testing framework, is installed along with all its dependencies. 
```

- Fix typo
- The second half of the sentence don't need pauses (using `,`)

```yaml build-and-test.yml
build:
  name: Build and test
  steps:
    - eas/build

>     # ... simulator / emulator setup

Reviewer: ```suggestion
    ... simulator/emulator setup
```

- Is it a requirement to mention emulator/simulator setup? I assume the following maestro job won't work with out it.

Author: What do you mean?

> I assume the following maestro job won't work with out it.

Indeed, without simulator/emulator set up `maestro test` will fail.

Reviewer: I had an idea that if we don't mention this step in the example snippet that maestro test might now work. Just wanted to confirm. Thank you!

    # @info #
    - eas/install_maestro:
    # @end #
        inputs:
          # @info If you need to, you can provide the Maestro version to install. #
          maestro_version: 1.35.0
          # @end #
    - run:
        command: maestro test flows/signin.yml
    - eas/upload_artifact:
        name: Upload Maestro artifacts
        inputs:
          type: build-artifact
          path: ${ eas.env.HOME }/.maestro/tests
```

| Input             | Description                                                                                                              |
|-------------------|--------------------------------------------------------------------------------------------------------------------------|

> | `maestro_version` | **string** Maestro version to install (e.g. 1.35.0). If not provided, `install_maestro` will install the latest version. |

Reviewer: ```suggestion
| `maestro_version` | **string** Maestro version to install (for example, 1.35.0). If not provided, `install_maestro` will install the latest version. |
```

- Let's avoid using latin abbreviations as per our [writing style guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md#abbreviations)

<BoxLink
  title="eas/install_maestro source code"
  description="View the source code for the eas/install_maestro function on GitHub."
  Icon={GithubIcon}
  href="https://github.com/expo/eas-build/blob/main/packages/build-tools/src/steps/functions/installMaestro.ts"
/>

#### `eas/start_android_emulator`

> Starts an Android emulator you can use to test your apps on. Only available when running a build for Android.

Reviewer: ```suggestion
Starts an Android Emulator you can use to test your apps on. Only available when running a build for Android.
```

        inputs:
          system_image_package: system-images;android-30;default;x86_64
    # @end #
    # ... Maestro setup and tests
```

| Input                  | Description                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               |
|------------------------|-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|

> | `device_name`          | **string** Name for the created device. You may wish to customize it if starting multiple emulators.                                                                                                                                                                                                                                                                                                                                                                                                      |

Reviewer: ```suggestion
| `device_name`          | **string** Name for the created device. You can customize it if starting multiple emulators.                                                                                                                                                                                                                                                                                                                                                                                                      |
```

> | `system_image_package` | **string** Android package path you wish to use for the emulator, e.g. `system-images;android-30;default;x86_64`.<br />To get list of available system images, run `sdkmanager --list` on your computer ([command docs](https://developer.android.com/tools/sdkmanager#list)). VMs run on x86_64 architecture, so always choose `x86_64` package variants. The `sdkmanager` tool comes from Android SDK command-line tools ([installation instructions](https://developer.android.com/tools/sdkmanager)). |

Reviewer: ```suggestion
| `system_image_package` | **string** Android package path to use for the emulator. For example, `system-images;android-30;default;x86_64`.<br />To get a list of available system images, run [`sdkmanager --list`](https://developer.android.com/tools/sdkmanager#list) on a local computer. VMs run on x86_64 architecture, so always choose `x86_64` package variants. The [`sdkmanager` tool](https://developer.android.com/tools/sdkmanager) comes from Android SDK command-line tools. |
```

- Let's highlight sdkmanager command and sdkmanager tools as links instead of explicitly specifying those docs links.

    - eas/start_ios_simulator
    # @end #
    # ... Maestro setup and tests
```

| Input               | Description                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       |
|---------------------|---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|

> | `device_identifier` | **string** Name or UDID of the Simulator you want to start. Examples include `iPhone [XY] Pro`, `AEF997BB-222C-4379-89BA-D21070B1D787`.<br />**Note:** available Simulators are different for every image, so if you change the image, the Simulator for a given name may become unavailable. For instance, an Xcode 14 image will have iPhone 14 Simulators while an Xcode 15 image will have iPhone 15 simulators. In general we encourage not providing this input. Learn more about [runner images](/build/eas-json/#selecting-a-base-image). |

Reviewer: ```suggestion
| `device_identifier` | **string** Name or UDID of the Simulator you want to start. Examples include `iPhone [XY] Pro`, `AEF997BB-222C-4379-89BA-D21070B1D787`.<br />**Note:** Available Simulators are different for every image. If you change the image, the Simulator for a given name may become unavailable. For instance, an Xcode 14 image will have iPhone 14 Simulators, while an Xcode 15 image will have iPhone 15 simulators. In general, we encourage not providing this input. See [runner images](/build/eas-json/#selecting-a-base-image) for more information.
```
